std/typecons.d

```diff
@@ -6208,3 +6208,109 @@ public:
     auto value = cast(int)flags_A;
     assert(value == Enum.A);
 }
+/**
+ * Returns a Tuple of the parameters.
+ * It can be used to declare function.
+ */
+template ParameterTuple(alias Func)
+{
+    import std.traits : FunctionTypeOf;
+    static if (is(FunctionTypeOf!Func Params == __parameters))
+       alias ParameterTuple = Params;
+    else
+       static assert(0, "Argument to ParameterTuple must be a function");
+}
+
+///
+unittest
+{
+    // There is a (probable) compiler bug, affecting variadic functions for
+    // dynamically loaded library. The first function call will have garbage arguments.
+    // @@@BUG@@@ 13769
+    void foo(string val = "Test", int = 10);
+    void bar(ParameterTuple!foo) { assert(val == "Test"); }
+    // Variadic functions require special handling:
+    import core.vararg;
+    void foo2(string val, ...);
+    void bar2(ParameterTuple!foo2/*, ...*/) { assert(val == "42", val); }
+
+    bar();
+    bar2("42");
+
+    // Works with parameter-less functions too.
+    void foo3();
+    void bar3(ParameterTuple!foo3) {}
+    bar3();
+
+    // Note: outside of a parameter list, it's value is the type of the param.
+    import std.traits : ParameterDefaultValueTuple;
+    ParameterTuple!(foo)[0] test = ParameterDefaultValueTuple!(foo)[0];
+    assert(test == "Test");
+}
+
+/**
+    Returns a Tuple containing a 1-element parameter list, with an optional default value.
+    Can be used to concatenate a parameter to a parameter list, or to create one.
+    Params:
+    T = Type of the parameter to declare.
+    identifier = identifier to give to the parameter. This can be an empty string.
+    defVal = Optional default value for the parameter. Passing void means no return value.
+	      This means that std.traits.ParameterDefaultValueTuple can be used directly.
+*/
+template ParameterTuple(T, string identifier, TUnused = void)
```

Comment:
Just remove `TUnused` and put a constraint on the overload with `defVal`: `if (!is(defVal))`